tests-beta/ios/MASVS-RESILIENCE/MASTG-TEST-0243.md

```diff
@@ -0,0 +1,25 @@
+---
+platform: ios
+title: References to Secure Lock Verification APIs
+id: MASTG-TEST-0243
+apis: [LAContext.canEvaluatePolicy, kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly]
+type: [static]
+weakness: MASWE-0008
+best-practices: []
+---
+
+## Overview
+
+This test verifies that an application is running on a device with a set passcode. A set passcode ensures that data on the device is encrypted and access to the device is restricted.
```

Comment:
```suggestion
Here's an enhanced version of the **iOS test overview**, incorporating the appropriate APIs:  

---

This test verifies that an app is running on a device with a secure lock screen (e.g. a passcode).  

On iOS, apps can determine whether a secure lock screen is set using the **LocalAuthentication** framework. Specifically, the `LAContext.canEvaluatePolicy(_:error:)` method with the `.deviceOwnerAuthentication` or `.deviceOwnerAuthenticationWithBiometrics` policy can be used to check if authentication mechanisms, including a passcode, are available.  

Apps leveraging the **Keychain Services API** can require passcode authentication before accessing sensitive data using the `kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly` attribute.  
```

Reply:
The suggestion is a bit broken, so I added it as a separate commit